extensions/iidm/include/powsybl/iidm/extensions/iidm/DiscreteMeasurement.hpp

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef POWSYBL_IIDM_EXTENSIONS_IIDM_DISCRETEMEASUREMENT_HPP
+#define POWSYBL_IIDM_EXTENSIONS_IIDM_DISCRETEMEASUREMENT_HPP
+
+#include <map>
+#include <string>
+
+#include <boost/any.hpp>
+
+#include <powsybl/stdcxx/optional.hpp>
+#include <powsybl/stdcxx/range.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+namespace extensions {
+
+namespace iidm {
+
+class DiscreteMeasurements;
+
+class DiscreteMeasurement {
+public:
+    /**
+     * Specify what is measured.
+     */
+    enum class Type : unsigned char {
+        TAP_POSITION,
+        SWITCH_POSITION,
+        SHUNT_COMPENSATOR_SECTION,
+        OTHER
+    };
+
+    /**
+     * If it is the modelization of a tap position (or any discrete measurement on a tap changer),
+     * can explicit which tap changer of the transformer it is applied on.
+     */
+    enum class TapChanger : unsigned char {
+        RATIO_TAP_CHANGER,
+        PHASE_TAP_CHANGER,
+        RATIO_TAP_CHANGER_1,
+        PHASE_TAP_CHANGER_1,
+        RATIO_TAP_CHANGER_2,
+        PHASE_TAP_CHANGER_2,
+        RATIO_TAP_CHANGER_3,
+        PHASE_TAP_CHANGER_3
+    };
+
+    /**
+     * Explicit what type of discrete value is used for this measurement (boolean, int or string).
+     */
+    enum class ValueType : unsigned char {
+        BOOLEAN,
+        INT,
+        STRING
+    };
+
+public:
+    DiscreteMeasurement(DiscreteMeasurements& discreteMeasurements, const std::string& id, const Type& type,
```

Comment:
To be discussed: is it really relevant event if this class is not an extension? #371 existed because it was dangerous to instantiate the extension without calling `Extendable::addExtension`. For this classe this is not relevant IMHO.

Reply:
The public API should not allow to create a `DiscreteMeasurement` instance without using the `DicreteMeasurementAdder`. The constructor should be declared private and the adder friend of this class.